File: square.cpp
#include <iostream>
#include <math.h>
using namespace std;
class Square{
  public: 
  int x1;
  int x2;
  int y1;
  int y2;
  int px;
  int py;
  
  
  
  Square(){
    x1 = 0;
    x2 = 3;
    y1 = 0;
    y2 = 3;
  }
  double calc_dist(){
      return sqrt(pow(x2 - x1, 2) + pow(y2 - y1, 2));
  }
  int dist_x(){
      return fabs(x2 - x1);
  }
  int dist_y(){
      return fabs(y2 - y1);
  }
  int area(){
      return dist_x() * dist_y();
  }
  int perimeter(){
      return 2 * (dist_x() + dist_y());
  }
  bool contains(int px, int py){
     bool contain = true;
      if(px > x1 && py > y1 || py == y1 && px < x2 || px == x2 && py < y2 || py == y2){
        return contain;
      }else{
        contain = false;
        return contain;
      }
  }
};
int main ()
{
  Square a;
  int punto_x;
  int punto_y;
    cin >> punto_x;
    cin >> punto_y;
    cout << a.perimeter()<< endl;
    cout << a.area() << endl;
    cout << a.contains(punto_x, punto_y);
    
  return 0;
}

File: square.py
import math
class Square:
    def __init__(square, first_x, scnd_x, first_y, scnd_y, px, py):
        square.first_x = first_x
        square.scnd_x = scnd_x
        square.first_y = first_y
        square.scnd_y = scnd_y
        square.px = px
        square.py = py
    
    def calc_dist(self):
        return int(math.sqrt(pow(self.scnd_x - self.first_y, 2) + pow(self.scnd_y - self.first_y, 2)))
    
    def dist_x(self):
        return abs(self.scnd_x - self.first_x)
        
    def dist_y(self):
        return abs(self.scnd_y - self.first_y)
    def area(self):
        return self.dist_x() * self.dist_y();
    def perimeter(self):
        return 2 * (self.dist_x() + self.dist_y())
    def contains(self):
        if(self.px >= self.first_x and self.py >= self.first_y and self.px <= self.scnd_x and self.py <= self.scnd_y):
            return True
        else:
            return False
        
      
s1 = Square(0, 3, 0, 5, 0, 10)

print(s1.calc_dist())
print(s1.dist_x())
print(s1.dist_y())
print(s1.area())
print(s1.contains())
